commit 49b3cf035edc5d7deb3ad1bf6805ce456ababc5b
Author: Peter Collingbourne <pcc@google.com>
Date:   Sat Nov 21 01:59:02 2020 -0800

    kasan: arm64: set TCR_EL1.TBID1 when enabled
    
    On hardware supporting pointer authentication, we previously ended up
    enabling TBI on instruction accesses when tag-based ASAN was enabled,
    but this was costing us 8 bits of PAC entropy, which was unnecessary
    since tag-based ASAN does not require TBI on instruction accesses. Get
    them back by setting TCR_EL1.TBID1.
    
    Signed-off-by: Peter Collingbourne <pcc@google.com>
    Reviewed-by: Andrey Konovalov <andreyknvl@google.com>
    Link: https://linux-review.googlesource.com/id/I3dded7824be2e70ea64df0aabab9598d5aebfcc4
    Link: https://lore.kernel.org/r/20f64e26fc8a1309caa446fffcb1b4e2fe9e229f.1605952129.git.pcc@google.com
    Signed-off-by: Catalin Marinas <catalin.marinas@arm.com>

diff --git a/arch/arm64/include/asm/pgtable-hwdef.h b/arch/arm64/include/asm/pgtable-hwdef.h
index 01a96d07ae74..42442a0ae2ab 100644
--- a/arch/arm64/include/asm/pgtable-hwdef.h
+++ b/arch/arm64/include/asm/pgtable-hwdef.h
@@ -260,6 +260,7 @@
 #define TCR_TBI1		(UL(1) << 38)
 #define TCR_HA			(UL(1) << 39)
 #define TCR_HD			(UL(1) << 40)
+#define TCR_TBID1		(UL(1) << 52)
 #define TCR_NFD0		(UL(1) << 53)
 #define TCR_NFD1		(UL(1) << 54)
 #define TCR_E0PD0		(UL(1) << 55)